### ISSUE TITLE
StyledSelect accessibility

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2056]
Reported here: https://twitter.com/WMassag/status/1130396510221492224

The `StyledSelect` component lacks accessibility. When typing the first letter(s) of the text it should behaves just like the native browser `select` and focus the most appropriate entry.

This is happening for the country select, but we should fix it at the component level to ensure all our existing and future selects will be covered.

**Comment by NotJohs** [comment]
@Betree can I get more information about this issue. 
PS: the twitter link is broken.

**Comment by Betree** [comment]
@NotJohs This issue should be fixed by our transition to `react-select` (see https://github.com/opencollective/opencollective/pull/2135). I'll soon open bounties to migrate the existing selects, stay tuned :wink:

**Comment by NotJohs** [comment]
@Betree alright

**Comment by Betree** [comment]
Closing this issue in favor of migrating to the new `StyledSelect` (that relies on react-select)

See #2292 #2293 #2294
